server/container/util/src/main/java/org/apache/james/util/concurrency/ConcurrentTestRunner.java

```diff
@@ -48,14 +54,19 @@ public ConcurrentRunnableTask(int threadNumber, BiConsumer biConsumer) {
 
         @Override
         public void run() {
+            exception = null;
             countDownLatch.countDown();
             for (int i = 0; i < operationCount; i++) {
                 try {
                     biConsumer.consume(threadNumber, i);
                 } catch (Exception e) {
                     LOGGER.error("Error caught during concurrent testing", e);
+                    exception = e;
```

Comment:
All the exceptions are LOGGED, with stacktrace. Thus you can fully correct your tests after one run.

Rethrowing the last exception is more for failing the tests. 

I don't buy the MultipleException.